python/solver.py

```diff
@@ -266,6 +257,24 @@ def _get_field(self, f, band, bloch_phase):
 
         return res
 
+    def get_curfield_as_array(self, bloch_phase=True):
+        dims = self.mode_solver.get_dims()
+        arr = np.zeros(np.prod(dims))
+        self.mode_solver.get_curfield(arr)
+        arr = np.reshape(arr, dims)
+
+        return MPBArray(arr, self.get_lattice(), self.current_k, bloch_phase=bloch_phase)
+
+    def get_dpwr(self, band):
+        self.get_dfield(band)
+        self.compute_field_energy()
+        return self.get_curfield_as_array()
```

Comment:
Shouldn’t Bloch_phase be false here?

Reply:
Yes, you're right. Fixed. I also fixed `get_tot_pwr`. It was multiplying by the Bloch phase too.